libraries/zookeeper_service.rb

```diff
@@ -86,6 +86,7 @@ class ZookeeperService < Chef::Provider
       include PoiseService::ServiceMixin
 
       def action_enable
+        new_resource.notifies(:restart, new_resource, :delayed)
```

Comment:
ahhhh ok. This is extremely useful if this is the case. I see across multiple poise developed cookbooks where notifies is being called directly in the provider block.